src/components/SchemaRow/useChoices.ts

```diff
@@ -0,0 +1,87 @@
+import { isReferenceNode, isRegularNode, SchemaNode } from '@stoplight/json-schema-tree';
+import { last } from 'lodash';
+import * as React from 'react';
+
+import { isComplexArray, isParentNode } from '../../tree';
+import { printName } from '../../utils';
+
+type Choice = {
+  title: string;
+  type: SchemaNode;
+};
+
+function calculateChoiceTitle(node: SchemaNode, isPlural: boolean): string {
+  const complexObjectSuffix = isPlural ? '-s' : '';
+  const primitiveSuffix = isPlural ? 's' : '';
+  if (isRegularNode(node)) {
+    const realName = printName(node);
+    if (realName) {
+      return realName + complexObjectSuffix;
+    }
+    return (node.primaryType ?? 'object') + primitiveSuffix;
```

Comment:
I'm not sure you want to default. If there's no associated `primaryType` you probably want to display `any` instead of `object`.
```suggestion
    return (node.primaryType ?? 'any') + primitiveSuffix;
```

Reply:
Good point, I'll correct.